Gems/RecastNavigation/Code/Source/Components/RecastNavigationPhysXProviderComponent.cpp

```diff
@@ -56,6 +58,14 @@ namespace RecastNavigation
         return CollectGeometryImpl(tileSize, borderSize, GetWorldBounds(), m_debugDrawInputData);
     }
 
+    void RecastNavigationPhysXProviderComponent::CollectGeometryAsync(
+        float tileSize,
+        float borderSize,
+        AZStd::function<void(AZStd::shared_ptr<TileGeometry>)> tileCallback)
```

Comment:
suggestion: As this AZStd::function is passed by value, we could `AZStd::move` it into the function below if it takes ownership (or we should pass by const ref)

Reply:
Sounds good. Added a move.